### PR TITLE
[#13954] Allows args and arg as schema properties in python client

### PR DESCRIPTION
**Pull request by MustansirS**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@spacether

### CODE REVIEW

File: docs/generators/python.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 <ul class="column-ul">
 <li>all_params</li>
 <li>and</li>
+<li>args</li>
```
Comment (spacether) [inline docs/generators/python.md:70]: Should arg also be added?
What do you think about instead converting all arg into _arg and args into _args
That way they can stay non-reserved?

Comment (MustansirS) [reply docs/generators/python.md:70]: Do you mean do these conversions in the templates?

Comment (spacether) [reply docs/generators/python.md:70]: Yes in the templates

**Review by spacether — COMMENTED** [review]
How about 
- also adding a component schema in the python petstore sample spec
  -  ObjectModelWithArgAndArgsProperties
- add a manual test in `tests_manual` that ensures that you can instantiate the model with those inputs. Maybe also assert that there are type hints for those properties?

```
components:
  schemas:
    ObjectModelWithArgAndArgsProperties:
      type: object
      additionalProperties:
        type: string
      required:
      - arg
      - args
```
Adding these suggested tests will ensure that this feature keeps working in the future, and that if someone removes it these suggested tests will fail.

**Comment by MustansirS** [comment]
> How about
> 
> * also adding a component schema in the python petstore sample spec
>   
>   * ObjectModelWithArgAndArgsProperties
> * add a manual test in `tests_manual` that ensures that you can instantiate the model with those inputs. Maybe also assert that there are type hints for those properties?
> 
> ```
> components:
>   schemas:
>     ObjectModelWithArgAndArgsProperties:
>       type: object
>       additionalProperties:
>         type: string
>       required:
>       - arg
>       - args
> ```
> 
> Adding these suggested tests will ensure that this feature keeps working in the future, and that if someone removes it these suggested tests will fail.

@spacether done 👍

**Review by spacether — COMMENTED** [review]
Thank you for the PR! This looks great. Only one more ask here. How about only changing the new and init method signatures to use _arg and _args?
All other signatures can stay as is.

When passing _arg and _args to other helper functions from those you can use arg=_arg args=_args
From_openapi_data doesn't need the underscores because it is always passed one arg input. So if arg and arg are passed into it, they are keys in an arg dict.

**Review by spacether — APPROVED** [review]
Thank you for your PR fixing this, and for making the suggested updates and adding test coverage. This looks great!

**Pull request merged**